packages/core/src/shared/clients/codecatalystClient.ts

```diff
@@ -175,7 +178,16 @@ export async function createClient(
 ): Promise<CodeCatalystClient> {
     const sdkClient = await createCodeCatalystClient(connection, regionCode, endpoint, retryOptions)
     const c = new CodeCatalystClientInternal(connection, sdkClient)
-    await c.verifySession()
+    try {
+        await c.verifySession()
+    } catch (e) {
+        if (!(e instanceof ToolkitError) || e.code !== 'AccessDeniedException') throw e
+        onAccessDeniedExceptionEmitter.fire()
+
+        // We "cancel" this error so that it prevents further execution,
+        // but doesn't display the error
+        throw new ToolkitError('CodeCatalyst scope is expired', { code: 'ScopeExpiration', cancelled: true })
```

Comment:
Is there a reason not to throw a `CancellationError`? https://github.com/aws/aws-toolkit-vscode/blob/2c4ea6516c5249c048b301d5826f4b0cfe16bf0e/packages/core/src/auth/sso/ssoAccessTokenProvider.ts#L188